### PR TITLE
stop_oncomplete_calling_on_forcestop_of_tween_if_already_completed

### PR DESCRIPTION
**Pull request by ChrisBindy**
fix: if the abstract animation is already finished no need to do anything so just return out of the function.

`MyAnimation?.Stop(true);` with an oncomplete function and stop it after it has finished playing.

### CODE REVIEW

File: Assets/FlowEnt/Scripts/Runtime/Core/Abstract/AbstractAnimation/AbstractAnimation.cs [hunk 1]
```diff
@@ -180,8 +180,9 @@ protected override void StopInternal(bool triggerOnCompleted)
             switch (playState)
             {
                 case PlayState.Building:
-                case PlayState.Finished:
                     break;
+                case PlayState.Finished:
```
Comment (danieltranca) [inline Assets/FlowEnt/Scripts/Runtime/Core/Abstract/AbstractAnimation/AbstractAnimation.cs:184]: can you update the animations tests to reflect the fact that this issue was fixed? (Calling Stop twice)

**Review by danieltranca — CHANGES_REQUESTED** [review]

File: Assets/FlowEnt/Scripts/Runtime/Core/Abstract/AbstractAnimation/AbstractAnimation.cs (hunk 1 above)
Comment (ChrisBindy) [reply Assets/FlowEnt/Scripts/Runtime/Core/Abstract/AbstractAnimation/AbstractAnimation.cs:184]: As in write a new test for it?

Comment (danieltranca) [reply Assets/FlowEnt/Scripts/Runtime/Core/Abstract/AbstractAnimation/AbstractAnimation.cs:184]: no, update the already existing stop test(there is an abstract generic one so you only need to do 1 for all 3 types of animation

Comment (danieltranca) [reply Assets/FlowEnt/Scripts/Runtime/Core/Abstract/AbstractAnimation/AbstractAnimation.cs:184]: call stop on assert again and check if it throws or not

**Review by danieltranca — APPROVED** [review]

**Pull request merged**